Repo setup

viniersiren opened the pull request:
Issues previously covered but not checked off:
Issue #1  Issue #4 

Issue #16 

Nearly Complete Navigation and Profile(Nav still requiring a custom solution for the back button between tabs)

staab commented on components/profile/profileCard.tsx:
```diff
@@ -0,0 +1,310 @@
+  }>;
+}
+
+export default function ProfileCard({
```
The way this is written, it looks like the idea is to make it generalizable. From experience, I can say that that will probably never happen. This is a profile form, and the only profile that can be edited by the current user is that of the current user. Which means you can save a lot of communication between components by making this entirely self-contained, with zero props. I see a lot of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) on the settings page, just `useStore(userProfile)` directly here.

Now, that said, it is sometimes worthwhile splitting out individual fields from a form for re-use, buffering pre-submit, and to keep your forms from becoming a sprawling mess. I see that ProfilePicture directly calls `changePicture`. I would actually put a handler on that component that it can use to pass its value to the caller, which can then implement event publishing immediately or on submit.

staab commented on components/theme/README.md:
```diff
@@ -0,0 +1,182 @@
+import { Text, View, useThemeColor } from '@/components/theme/Themed';
+
+function MyComponent() {
+  const backgroundColor = useThemeColor({}, 'background');
```
I would reverse the order of arguments, and make the parameters optional

staab commented on utils/dataHandling.ts:
```diff
@@ -0,0 +1,277 @@
+ * Establishes initial relay connections on app launch
+ * @returns Promise that resolves when connections are established
+ */
+export const establishRelayConnections = async (): Promise<void> => {
```
This shouldn't really be necessary, connections are pretty quick, and should be managed mostly automatically. If a connection fails, welshman will automatically fall back on subsequent requests.

staab commented on utils/dataHandling.ts:
```diff
@@ -0,0 +1,277 @@
+    console.log('[PROFILE] Loading profile for pubkey:', pubkey);
+    
+    const router = Router.get();
+    let readRelays = router.ForPubkey(pubkey).getUrls();
```
This is actually backwards, ForPubkey is used to select relay the user _reads notes created by other people that mention him from_. You want `FromPubkey`, which is the relays the user writes his own notes to. Since profile is created by the user, that's where it will go.

staab commented on utils/dataHandling.ts:
```diff
@@ -0,0 +1,277 @@
+ * @param pubkey - The user's public key
+ * @returns Promise that resolves with the profile data
+ */
+export const loadUserProfile = async (pubkey: string): Promise<any> => {
```
This whole function can be replaced with `useStore(userProfile)` (userProfile is in welshman/app)

staab commented on utils/dataHandling.ts:
```diff
@@ -0,0 +1,277 @@
+    }
+    console.log('[PROFILE] Current profile data:', profileData);
+
+    const profileContent = JSON.stringify({
```
See @welshman/util -> Profile for an `editProfile` and `Profile` type which can help with this.

staab commented on utils/dataHandling.ts:
```diff
@@ -0,0 +1,277 @@
+    const router = Router.get();
+    const publishRelays = router.FromUser().getUrls();
+    
+    if (publishRelays.length === 0) {
```
This will never happen, since the router automatically falls back to the default relays.

staab commented on utils/dataHandling.ts:
```diff
@@ -0,0 +1,277 @@
+ * @param currentProfile - Optional current profile data to avoid reloading
+ * @returns Promise that resolves when the event is published
+ */
+export const changeDescription = async (newDescription: string, currentProfile?: any): Promise<void> => {
```
Because this is so similar to changeName, and because `Profile` is a known type, it's best to just do a single function to publish the event. I've added this to welshman as `setProfile` (although we'll eventually have to add a custom version like the one in flotilla).

staab commented on utils/imageUpload.ts:
```diff
@@ -0,0 +1,214 @@
+    // On mobile, use the local URI for Nostr storage
+    return imageResult.uri;
+  }
+}; 
```
Most of this file looks good for dealing with the platform, just decouple that from the actual uploading code, which can go elsewhere.

staab commented on utils/useClientOnlyValue.ts:
```diff
@@ -0,0 +1,4 @@
+// This function is web-only as native doesn't currently support server (or build-time) rendering.
+export function useClientOnlyValue<S, C>(server: S, client: C): S | C {
```
What is this?

viniersiren replied:
These were generated by the Expo Project setup. I didn't really understand it so I kept it in. It looks like I can use it for conditionally loading instead of code like ```const isWeb = Platform.OS === 'web';``` since that is how it is currently used to load the web-only topbar.

viniersiren commented:
I added the majority of your suggested changes, especially the welshman functions that replaced/worked better than my manual ones. I'm going to split the imageUpload.ts file as an example of using the proposed file structure. I'll also add in some of my custom Routing logic from a different branch and complete so the back button goes across tabs. 

The tests were deleted; They were basically leftover code from my initial exploration of Welshman. I kept the testing folders but I am unsure when/if I will make comprehensive tests. Also whitespaces should be trimmed, I changed my editor settings then ran an eslint script which was not included in the commit. 

I had some difficulty with the CSP adding this black bar on the top so it is commented out. I don't really know what I'm doing with the CSP so
